mono/mini/mini-runtime.c

```diff
@@ -351,11 +351,18 @@ void *mono_global_codeman_reserve (int size)
 	if (!global_codeman) {
 		/* This can happen during startup */
 		global_codeman = mono_code_manager_new ();
+
+		MONO_SCOPE_ENABLE_JIT_WRITE();
 		return mono_code_manager_reserve (global_codeman, size);
 	}
 	else {
 		mono_jit_lock ();
-		ptr = mono_code_manager_reserve (global_codeman, size);
+
+		{
```

Comment:
Why does this need done in the braces?

Reply:
I didn't want to extend the mode switch past `mono_code_manager_reserve`, as I don't know if any other calls in there depend the JIT being in executable mode.